### ISSUE TITLE
X was supposed to be in Y.cmt but could not be found

### ISSUE THREAD

**Issue by scg03** [ocaml/merlin#648]
I was trying to read some lablgtk source code.
MERLIN_LOG shows:
# 0.01 dot_merlin - update filenames
[ "/home/ocaml/lablgtk-2.18.5/build/src" ]
# 0.01 dot_merlin - update files
[ "/home/ocaml/lablgtk-2.18.5/build/.merlin" ]
# 0.01 frontend - output
{ "class": "return", "value": true, "notifications": [] }
# 0.01 frontend - input
{
  "document": [ "auto", "/home/ocaml/lablgtk-2.18.5/build/src/gTree.ml" ],
  "query": [ "locate", null, "ml", "at", { "line": 108, "col": 20 } ]
}
# 0.06 track_definition - from_string
looking for the source of 'tree_sortable_sigs' (prioritizing .ml files)
# 0.06 track_definition - lookup
lookup in value namespace
# 0.06 track_definition - lookup
lookup in type namespace
# 0.06 track_definition - locate
present in the environment, but ghost lock.
walking up the typedtree looking for 'OgtkTreeProps.tree_sortable_sigs[type]'
# 0.06 typedtrie - ignored node
class_declaration
# 0.06 typedtrie - ignored node
class_declaration
# 0.06 typedtrie - ignored node
class_declaration
# 0.06 typedtrie - ignored node
class_declaration
# 0.06 typedtrie - ignored node
class_declaration
# 0.06 typedtrie - ignored node
open_description
# 0.06 typedtrie - ignored node
open_description
# 0.06 typedtrie - ignored node
open_description
# 0.06 typedtrie - ignored node
open_description
# 0.06 typedtrie - ignored node
open_description
# 0.06 typedtrie - ignored node
open_description
# 0.06 typedtrie - ignored node
open_description
# 0.06 typedtrie - ignored node
open_description
# 0.06 typedtrie - ignored node
open_description
# 0.06 typedtrie - ignored node
open_description
# 0.06 track_definition - locate
resolves to OgtkTreeProps.tree_sortable_sigs[type]
# 0.06 track_definition - Fallback.setopt
None
# 0.06 track_definition - from_path '%s'
OgtkTreeProps.tree_sortable_sigs[type]
# 0.06 track_definition - browse_cmts
inspecting /home/ocaml/lablgtk-2.18.5/build/src/ogtkTreeProps.cmt
# 0.06 track_definition - File_switching.move_to
/home/ocaml/lablgtk-2.18.5/build/src/ogtkTreeProps.cmt
# 0.06 typedtrie - ignored node
class_declaration
# 0.06 typedtrie - ignored node
class_declaration
# 0.06 typedtrie - ignored node
class_declaration
# 0.06 typedtrie - ignored node
class_declaration
# 0.06 typedtrie - ignored node
class_declaration
# 0.06 typedtrie - ignored node
class_declaration
# 0.06 typedtrie - ignored node
class_declaration
# 0.06 typedtrie - ignored node
class_declaration
# 0.06 typedtrie - ignored node
class_declaration
# 0.06 typedtrie - ignored node
class_declaration
# 0.06 typedtrie - ignored node
open_description
# 0.06 typedtrie - ignored node
open_description
# 0.06 typedtrie - ignored node
open_description
# 0.06 typedtrie - ignored node
open_description
# 0.06 track_definition - locate
new path (tree_sortable_sigs[type]) is not a real path. fallbacking...
# 0.06 track_definition - locate
{
cell_renderer_param -> (val) File "ogtkTreeProps.ml", line 120, characters 4-23
cell_renderer_pixbuf_param -> (val) File "ogtkTreeProps.ml", line 135, characters 4-30
cell_renderer_progress_param -> (val) File "ogtkTreeProps.ml", line 177, characters 4-32
cell_renderer_text_param -> (val) File "ogtkTreeProps.ml", line 145, characters 4-28
cell_renderer_toggle_param -> (val) File "ogtkTreeProps.ml", line 171, characters 4-30
get -> (val) File "ogtkTreeProps.ml", line 5, characters 4-7
param -> (val) File "ogtkTreeProps.ml", line 6, characters 4-9
set -> (val) File "ogtkTreeProps.ml", line 4, characters 4-7
}

# 0.06 frontend - output
{
  "class": "return",
  "value":
    "tree_sortable_sigs was supposed to be in /home/ocaml/lablgtk-2.18.5/build/src/ogtkTreeProps.cmt but could not be found",
  "notifications": []
}

**Comment by let-def** [comment]
Did you build with `-bin-annot`?

**Comment by scg03** [comment]
Yes. The log shows:
`ocamlc.opt  -g -dtypes -bin-annot -w s-3+52 -warn-error A-52 -c  ogtkTreeProps.ml`

and the OgtkTreeProps.cmt can be found in src directory.

**Comment by let-def** [comment]
Ah... I am note sure locate handles class types. (@trefis?)

**Comment by scg03** [comment]
Yes. I noticed merlin fails at class. In this case, I was trying to locate the class in the "inherit" clause and it failed (It has succeed in some other cases before). I don't know the internal details of merlin but I think it might not be difficult to handle locating the inherited class.

Thanks.

**Comment by trefis** [comment]
Yes indeed, classes, and class types, are not handled (confirmed by [these lines](https://gist.github.com/scg03/b454b8f523cf91fe98c2980f631d161e#file-merlin-log-L21) in the log).

Adding support shouldn't be too hard. I'll try and do that soon™
